### ISSUE TITLE
documentation corrections

### ISSUE THREAD

**Issue by adalke** [pallets/click#1778]
Here are a three corrections to the click documentation

1) The example at https://click.palletsprojects.com/en/7.x/options/#password-prompts uses:

```
click.echo('Encrypting password to %s' % password.encode('rot13'))
```

That is Python 2-specific. I believe the correct Python 3 version (portable to Python 2) is:
```
import codecs
codecs.encode(password, "rot13")
```
2) The example at https://click.palletsprojects.com/en/7.x/options/#dynamic-defaults-for-prompts uses ``lambda: os.environ.get ...`.

It missing an `import os`. When I copy&pasted to follow along, it gave me a NameError for os. Note that elsewhere there are two `import sys` statement so the documentation appears to want to be complete about which modules are needed.

3) The example at https://click.palletsprojects.com/en/7.x/options/#callbacks-for-validation uses
```
        rolls, dice = map(int, value.split('d', 2))
```
While not a bug that causes an observable difference, that 2 should be a 1 as it indicates the number of splits that should be performed, not the number of terms to split into.

Environment:

- Python version: N/A
- Click version: 7.x

**Comment by davidism** [comment]
1. submit a PR
2. we don't add imports everywhere as it gets too verbose and isn't relevant to what the example is demonstrating
3. submit a PR

Generally, you don't need to open an issue to discuss docs, just open a PR.

**Comment by adalke** [comment]
Feel free to close this. I clearly don't know what I'm doing here, and have given up.

That is, I cloned the repo to so I could test the edits, founded it needed a newer version of Python to build the docs. Switched to a 3.9 install then found it needed pallets_sphinx_themes. Installed that, then found it needed sphinxcontrib.log_cabinet. Installed that, then found it needed sphinx_issues.

Only after doing that did I see there's a "CONTRIBUTING.rst" with those details.

On top of that, git and GitHub confuse me, so I gave up.
